### PR TITLE
Migrate to react-leaflet v3 (redo)

### PR DESCRIPTION
**Pull request by MasterEric**
Had to recreate the pull request since I realized the original PR was trying to merge the wrong branch.

Compared to #49:

* getOptions has been cleared away, making the code related to parameters simpler.
* I removed the component test since react-leaflet no longer uses the components as children.
* I added a note to the test that was failing because of PaulLeCam/react-leaflet#830.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,4 +1,74 @@
-import { withLeaflet } from 'react-leaflet';
-import TextPath from './TextPath';
+import { createPathComponent } from '@react-leaflet/core';
+import { Polyline as LeafletPolyline } from 'leaflet';
+import 'leaflet-textpath';
 
-export default withLeaflet(TextPath);
+// eslint-disable-next-line no-unused-vars
```
Comment (clementallen) [inline src/index.js:5]: Linting passes for me with this comment removed

File: src/index.js [hunk 2]
```diff
@@ -1,4 +1,74 @@
...
+        below,
+        offset,
+        orientation,
+        attributes,
+        // PolyLine options.
+        ...options
+    },
+    ctx
+) => {
+    // const { positions, options, text, ...pathOptions } = getOptions(props);
```
Comment (clementallen) [inline src/index.js:23]: Can this be removed please as no longer needed?

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,74 @@
...
+        center,
+        below,
+        offset,
+        orientation,
+        attributes,
+    });
+    return { instance, context: { ...ctx, overlayContainer: instance } };
+};
+
+// eslint-disable-next-line no-unused-vars
```
Comment (clementallen) [inline src/index.js:36]: Linting passes for me with this comment removed

**Review by clementallen — CHANGES_REQUESTED** [review]
Looks good!  Just a few unused lines that can be removed.

I'm keen to not merge this to master until the issue around updating the component has been resolved.  I can either keep the MR open or merge it to a separate branch now and make any further changes myself later on, what would you prefer?

**Comment by MasterEric** [comment]
> I can either keep the MR open or merge it to a separate branch now and make any further changes myself later on, what would you prefer?

I think the latter makes more sense to me (merging into a branch pending the upstream fix).

Thanks for your help in reviewing this PR! It's greatly appreciated. I hope the issue gets resolved soon.

**Review by clementallen — APPROVED** [review]

**Comment by clementallen** [comment]
All looks good to me, thanks again for helping with this @MasterEric!

I'll update #48 once this is published pending any potential changes in react-leaflet itself

**Comment by nemosmithasf** [comment]
I'm using this version right now, and it seems like the `below` attribute doesn't work.

**Comment by clementallen** [comment]
Hi @nemosmithasf - I've just tested the `below` parameter locally and it works fine with this version.  Can you share some example code of how you're using the library and what you're expecting?

**Comment by nemosmithasf** [comment]
Here's the code I'm using, I'm expecting the text to switch between the top/bottom of the line when I toggle "shouldFlipText" (similar to if I multiplied offset by -1)
```

    const bearing = calcBearing(currCoordinate, endCoordinate);
    const shouldFlipText = 0 > bearing;

    return (
      <TextPath_baseComponent
        positions={[[currCoordinate, endCoordinate]]}
        color={"transparent"}
        text={text}
        center
        offset={20}
        below={shouldFlipText} 
        orientation={shouldFlipText ? "flip" : 0}
        attributes={{
          "font-size": fontSize,
          opacity: textOpacity,
          fill: fontColor,
          "font-weight": fontWeight,
          ...attributes,
        }}
      />
    );
```

**Comment by clementallen** [comment]
@nemosmithasf the `below` attribute actually controls whether the text sits below or above the line in terms of z-index, not positioning.  To achieve what you're wanting you can set the offset to either be `-20` or `20` I think

**Comment by clementallen** [comment]
I wasn't going to publish the new v3 support to NPM until PaulLeCam/react-leaflet#830 was resolved, but if it would help you I can publish it as a beta version?

**Comment by nemosmithasf** [comment]
Ah, of course, when you put it that way, it makes a lot of sense.

As for v3; it depends on how much work it's for you; I've hacked together a working version so I can just set a reminder to check back in a few weeks when you're fully satisfied with the v3 version. No point adding extra work for you unless it's trivial.

**Comment by clementallen** [comment]
@nemosmithasf I agree, it is a bit confusing - took me a few minutes to remember what it did myself!

No problem at all to publish a pre-release version, I've done it now.  You can install by running `npm install --save react-leaflet-textpath@next`

Hope that helps

**Comment by nemosmithasf** [comment]
Appreciate it, and thanks for the good work

**Comment by clementallen** [comment]
@nemosmithasf - no luck in the issue getting resolved in react-leaflet so will just have to accept it!  I've now published the v3 support as `2.0.0`

**Comment by nemosmithasf** [comment]
thanks!

**Pull request merged**